### ISSUE TITLE
Unexpected --cache Behavior with --list-different

### ISSUE THREAD

**Issue by rwaskiewicz** [prettier/prettier#13015]
**Environments:**

- Prettier Version: 2.7.1
- Running Prettier via: `npx` commands
- Runtime: Node v16.14.0, npm v8.3.1
- Operating System: macOS
- Prettier plugins (if any): none
**Steps to reproduce:**

Minimal Reproduction: https://github.com/rwaskiewicz/prettier-caching
Steps to Reproduce can be found in the `README` at the root of the repository

**Expected behavior:**

Running the following commands in sequence:
```
npx prettier . --list-different --cache
npx prettier . --write --cache
```

should format a project's code.

**Actual behavior:**

Calling `npx prettier . --list-different --cache` writes a cache. Calling `npx prettier . --write --cache` afterwards uses that cache and does not format/write the changes to disk.

**Additional Information**

This caught me off guard more than anything.  The workaround (or perhaps the desired behavior) would be to not run the first command with `--cache` like so:
```
npx prettier . --list-different
npx prettier . --write --cache
```

Alternatively, I can see the perspective of `--list-different --cache` not being allowed to use together.

**Comment by DanielSWolf** [comment]
This isn't the only case where the cache doesn't work as expected. If I run `prettier --check --cache .` more than once, the first run will actually print the modified files, as expected. The second run and any further runs, however, will wrongly print "All matched files use Prettier code style!"

It seems to me that the current behavior of the cache is broken.

**Comment by ksandvik** [comment]
I also saw that --cache worked fine, then upgraded to latest node 18.4.0 and cache does not work on first run in a  newly directory (used to work before), now I need to runs to get the cache activated.

**Comment by HolgerJeromin** [comment]
I would love to have this fixed for 3.0 (just checked with `3.0.0-alpha.0` and got hit by the bug)
We even have a PR ready #13016 thanks to @Milly 

cc @fisker

**Comment by rwaskiewicz** [comment]
Closing, as the bug appears to have been resolved in https://github.com/prettier/prettier/pull/13016 and is available in [Prettier v2.8.0](https://github.com/prettier/prettier/releases/tag/2.8.0). Thanks to @Milly for the fix and the Prettier team for their work! ❤️
